### DISCUSSION TITLE
Possible to Embed Game Engine: Godot as a 3D Control?

### DISCUSSION THREAD

**Discussion by GeorgeS2019** [AvaloniaUI/Avalonia#12709] [Ideas]
Godot now support DirectX12, Vulkan and OpenGL.
It is now available as an embedded library. .NET6/AOT.  => [LibGodotSharp](https://github.com/RhubarbVR/LibGodotSharp), https://github.com/RhubarbVR/LibGodotSharp/issues/4

see [Godot bindings for Swift](https://github.com/godotengine/godot-proposals/discussions/6920) - championed by **MONO Creator:** Miguel Deicaza

**Comment by wieslawsoltes** [comment]
> Godot now support DirectX12, Vulkan and OpenGL. It is now available as an embedded library. .NET6/AOT. => [LibGodotSharp](https://github.com/RhubarbVR/LibGodotSharp), [RhubarbVR/LibGodotSharp#4](https://github.com/RhubarbVR/LibGodotSharp/issues/4)
> 
> see [Godot bindings for Swift](https://github.com/godotengine/godot-proposals/discussions/6920) - championed by **MONO Creator:** Miguel Deicaza

https://github.com/MrJul/Estragonia

**Reply by GeorgeS2019** [reply]
@wieslawsoltes 

With Avalonia, is it possible to use XAML editor like Blend for VS2022 seamlessly?

**Reply by timunie** [reply]
probably not as Avalonia has it's own dialect. Some adjustments may be needed. Blend saves WPF dialect iirc. Even MAUI cannot use Blend afaik.

**Reply by timunie** [reply]
Inkscape can export Avalonia-XAML. but I don't know how good that is though.

**Reply by wieslawsoltes** [reply]
There is xaml editor with previewer (not designer) support for VS 2022 and Rider, no Blend support.

**Reply by GeorgeS2019** [reply]
Since [UrhoSharp](https://github.com/xamarin/urho) advocated by Xamarin, the search for 3D control for XAML has never being fulfilled.  Godot is the most complete 3D game engine MIT and top 8th Ranked in MIT license for entire Github. 

Now with Avalonia, perhaps the end of search is closer :-)

Godot vision is always Intuitive and super low learning curve. 

The question is how to make Avalonia as the unmatched Godot UI designer with Godot serving as embedded 3D Control?
